transports/quic/src/endpoint.rs

```diff
@@ -402,18 +402,41 @@ where
 
         if me.event_slot.is_none() && me.incoming_slot.is_none() {
             let mut metas = [RecvMeta::default(); BATCH_SIZE];
-            let mut iovs = MaybeUninit::<[IoSliceMut; BATCH_SIZE]>::uninit();
-            me.recv_buf
-                .chunks_mut(me.recv_buf.len() / BATCH_SIZE)
-                .enumerate()
-                .for_each(|(i, buf)| unsafe {
-                    iovs.as_mut_ptr()
-                        .cast::<IoSliceMut>()
-                        .add(i)
-                        .write(IoSliceMut::new(buf));
-                });
-            let mut iovs = unsafe { iovs.assume_init() };
-            while let Poll::Ready(result) = me.socket.poll_recv(cx, &mut iovs, &mut metas) {
+            let mut iovs = unsafe {
+                // TODO: use MaybeUninit::uninit_array when stabilized
+                // SAFETY: An uninitialized `[MaybeUninit<_>; LEN]` is valid.
+                MaybeUninit::<[MaybeUninit<IoSliceMut>; BATCH_SIZE]>::uninit().assume_init()
+            };
```

Comment:
This is wrong. It is not valid to assume uninitialized bytes are initialized.

Reply:
It is valid: https://doc.rust-lang.org/stable/src/core/mem/maybe_uninit.rs.html#353-356

Reply:
mmh, ok